File: lib/plugins/coin.js
module.exports = function(bot, config) {

  bot.respond(/(throw|flip|toss)( a)? coin/i, function(msg, reply) {
    var i = Math.floor(Math.random() * 2);
    if (i == 0) {
      reply("heads");
    } else {
      reply("tails");
    }
  });

  bot.help("coin", function(msg, reply) {
    reply("Throws coins");
    reply("Syntax: throw|flip|toss [a] coin");
  });

};

File: plugins/coin.py
import random

class Plugin:
    def __call__(self, bot):
        bot.on_respond(r"(throw|flip|toss)( a)? coin", self.on_respond)
        bot.on_help("coin", self.on_help)

    def on_respond(self, bot, msg, reply):
        reply(random.choice([ "Heads", "Tails" ]))

    def on_help(self, bot, msg, reply):
        reply("Throws coins.")
        reply("Syntax: throw|flip|toss [a] coin")
